### PR TITLE
Enable PixRacer onboard LED

### PR DESCRIPTION
**Pull request by iseries1**
Enable PixRacer onboard LED to show status of flight controller

### CODE REVIEW

File: libraries/AP_Notify/PixRacerLED.cpp [hunk 1]
```diff
@@ -0,0 +1,266 @@
...
+   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+   GNU General Public License for more details.
+
+   You should have received a copy of the GNU General Public License
+   along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+#include "PixRacerLED.h"
+#include <AP_GPS/AP_GPS.h>
+#include "AP_Notify.h"
+
```
Comment (lucasdemarchi) [inline libraries/AP_Notify/PixRacerLED.cpp]: Is it that different from other PX* boards that it needs a separate class?

File: libraries/AP_HAL_PX4/GPIO.cpp [hunk 2]
```diff
@@ -256,7 +266,24 @@ void PX4GPIO::write(uint8_t pin, uint8_t value)
 
 void PX4GPIO::toggle(uint8_t pin)
 {
-    write(pin, !read(pin));
+    switch (pin) {
+
+#ifdef CONFIG_ARCH_BOARD_PX4FMU_V4
+        case HAL_GPIO_A_LED_PIN:    // Arming LED
+                ioctl(_led_fd, LED_TOGGLE, LED_RED);
```
Comment (lucasdemarchi) [inline libraries/AP_HAL_PX4/GPIO.cpp]: I can't see where LED_RED is defined.... but you should define it to HAL_GPIO_A_LED_PIN for pixracer so you don't need the switch/case here.. There are also some minor coding style issues: ioctl() here should be one indentation less.

File: libraries/AP_Notify/PixRacerLED.cpp (hunk 1 above)
Comment (iseries1) [inline libraries/AP_Notify/PixRacerLED.cpp]: I tried without using this class and the LED's just don't work correctly and are hard to read.  So created this class to handle the light show better.

File: libraries/AP_HAL_PX4/GPIO.cpp (hunk 2 above)
Comment (iseries1) [inline libraries/AP_HAL_PX4/GPIO.cpp]: Didn't really do this code.  This was copied from 182... above.

**Comment by rmackay9** [comment]
I've created a new PR which re-uses some of the commits from this PR but modifies the PixRacerLED class to be a child of RGBLED.  This greatly reduces the amount of code required and I think it's the right way to do it.  The commits carrier over to the new PR still have iseries as the author.

**Pull request closed without merging**